Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -1078,31 +1078,18 @@ - (void)_updateOverlayRegions:(const WebKit::RemoteLayerTreeTransaction::LayerPr
     const auto& fixedIDs = scrollingCoordinatorProxy->fixedScrollingNodeLayerIDs();
 
     auto overlayRegionsWithIDs = layerTreeHost.overlayRegionsWithIDs();
-    auto initialOverlayRegionsWithIDsSize = overlayRegionsWithIDs.size();
     for (auto layerID : destroyedLayers)
         overlayRegionsWithIDs.remove(layerID);
 
-    bool didChangeOverlayRegions = overlayRegionsWithIDs.size() != initialOverlayRegionsWithIDsSize;
-
     for (auto layerID : fixedIDs)
-        addOverlayEventRegions(layerID, changedLayerPropertiesMap, overlayRegionsWithIDs, didChangeOverlayRegions);
-
-    if (didChangeOverlayRegions) {
-        [_scrollView willChangeValueForKey:@"overlayRegions"];
-        layerTreeProxy.updateOverlayRegionsWithIDs(overlayRegionsWithIDs);
-        [_scrollView didChangeValueForKey:@"overlayRegions"];
-    }
-}
-
-- (NSArray<NSData *> *)_overlayRegions
-{
-    const auto& layerTreeHost = downcast<WebKit::RemoteLayerTreeDrawingAreaProxy>(*_page->drawingArea()).remoteLayerTreeHost();
+        addOverlayEventRegions(layerID, changedLayerPropertiesMap, overlayRegionsWithIDs, layerTreeHost, self);
 
-    NSMutableArray<NSData *> *overlayRegions = [NSMutableArray array];
-    for (const auto& [layerID, rect] : layerTreeHost.overlayRegionsWithIDs())
-        [overlayRegions addObject:[NSData dataWithBytes:(void*)&rect length:sizeof(rect)]];
+    Vector<CGRect> overlayRegions;
```

Comment:
I could make `-[WKScrollView _updateOverlayRegions:]` take iterators to avoid building this vector, but it complicates the implementation slightly and the vector size is often only a few elements.